### DISCUSSION TITLE
Batch Training and DataLoader in PyG

### DISCUSSION THREAD

**Discussion by khaled-rahman** [pyg-team/pytorch_geometric#3154] [Q&A] [answered]
Hi, I was trying to train a model in PyTorch Lightning for large graphs.
To create a batch, I used `NeighborSampler` following pl [examples](https://github.com/pyg-team/pytorch_geometric/blob/ffaf0922d0d1cdb85a89084d759e61ae5735250e/examples/pytorch_lightning/graph_sage.py).
To create the batch graph, is it possible to take all neighbors instead of size parameters?
More specifically, I would like to create the batch on nodes, e.g., if node_ids = [0, 1, 2, 3, 4, ..., 49], and batch_size=4, then I would like to take node ids [0, 1, 2, 3] in a batch, [4, 5, 6, 7] in the second batch and so on. Also, the edge_index for the first batch should contain all the edges of a graph that are incoming/outgoing from the nodes [0, 1, 2, 3], i.e., `Data(x = [0, 1, 2, 3], edge_index = [[0, 1, 2, 2, 3,. ...], [2, 3, 5, 6, 0, ...]])`

Is it possible to create such batches using `NeighborSample` or `NeighborLoader`?
It would be helpful to get some pointers on this.

**Comment by wsad1** [comment, marked as answer]
Setting `sizes=-1` will get you all neighbors of a node in  `NeighborSampler` as explained [here](https://pytorch-geometric.readthedocs.io/en/latest/modules/loader.html#torch_geometric.loader.NeighborSampler).

Now  `NeighborSampler` is depreciated so its better to use `NeighborLoader`. In `NeighborLoader` although its not mentioned in the docs, and I am not entirely sure, it looks like you can get all neighbors by setting `num_neigbhors=[-1]`, @rusty1s can confirm this.

**Reply by khaled-rahman** [reply]
Thank you so much!

**Reply by rusty1s** [reply]
Added the missing piece of information to the doc, thanks @wsad1 :)
